File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/documents/core/implementation/DocumentServiceTest.java
package io.zenwave360.example.clinicaltool.modules.documents.core.implementation;

import io.zenwave360.example.clinicaltool.modules.documents.config.*;
import io.zenwave360.example.clinicaltool.modules.documents.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.documents.core.inbound.*;
import io.zenwave360.example.clinicaltool.modules.documents.core.inbound.dtos.*;
import io.zenwave360.example.clinicaltool.modules.documents.core.implementation.mappers.*;
import io.zenwave360.example.clinicaltool.modules.documents.core.outbound.jpa.*;
import io.zenwave360.example.clinicaltool.modules.documents.infrastructure.jpa.inmemory.*;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mapstruct.factory.Mappers;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.PageRequest;

import java.util.Map;
import java.util.Optional;
import java.time.*;
import java.math.BigDecimal;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.mockito.Mockito.*;

/**
 * Acceptance Test for DocumentService.
 */
class DocumentServiceTest  {

    private final Logger log = LoggerFactory.getLogger(getClass());

    ServicesInMemoryConfig context = new ServicesInMemoryConfig();
    DocumentServiceImpl documentService = context.documentService();

    DocumentInfoRepositoryInMemory documentInfoRepository = context.documentInfoRepository();

	@BeforeEach
	void setUp() {
		context.reloadTestData();
	}



    @Test
    void listDocumentInfoTest() {// TODO: implement this test
}

    @Test
    void downloadDocumentTest() {// TODO: implement this test
}

    @Test
    void uploadDocumentTest() {// TODO: implement this test
}

    @Test
    void deleteDocumentInfoTest() {
        Long id = null;
        // assertTrue(documentInfoRepository.containsKey(id));
        documentService.deleteDocumentInfo(id);
        // assertFalse(documentInfoRepository.containsKey(id));
}

}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/documents/DocumentServiceTest.kt
package io.zenwave360.example.clinicaltool.modules.documents

import io.zenwave360.example.clinicaltool.modules.documents.*
import io.zenwave360.example.clinicaltool.modules.documents.config.*
import io.zenwave360.example.clinicaltool.modules.documents.domain.*
import io.zenwave360.example.clinicaltool.modules.documents.dtos.*
import io.zenwave360.example.clinicaltool.modules.documents.inmemory.*
import io.zenwave360.example.clinicaltool.modules.documents.mappers.*
import java.time.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.Mockito.*
import org.slf4j.Logger
import org.slf4j.LoggerFactory

/** Acceptance Test for DocumentService. */
class DocumentServiceTest {

    private val log: Logger = LoggerFactory.getLogger(javaClass)

    val context = ServicesInMemoryConfig()
    val documentService: DocumentService = context.documentService()

    val documentInfoRepository: DocumentInfoRepositoryInMemory = context.documentInfoRepository()

    @BeforeEach
    fun setUp() {
        context.reloadTestData()
    }

    @Test
    fun listDocumentInfoTest() { // TODO: implement this test
    }

    @Test
    fun downloadDocumentTest() { // TODO: implement this test
    }

    @Test
    fun uploadDocumentTest() { // TODO: implement this test
    }

    @Test
    fun deleteDocumentInfoTest() {
        val id: Long = 1L
        // assertTrue(documentInfoRepository.containsKey(id))
        documentService.deleteDocumentInfo(id)
        // assertFalse(documentInfoRepository.containsKey(id))
    }
}
